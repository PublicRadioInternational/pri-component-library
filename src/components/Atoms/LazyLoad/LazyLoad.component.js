/**
 * @file LazyLoad.component.js
 * Exports a lazy load component.
 */

import 'intersection-observer'; // optional polyfill
import Observer from '@researchgate/react-intersection-observer';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Component that renders a Card Item.
 */
export default class LazyLoad extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired
  };

  loadImage = item => {
    if (item.isIntersecting) {
      const element = item.target;
      const { src } = element.dataset;
      element.src = src;
      element.addEventListener('load', () => {
        element.removeAttribute('data-src');
      });
    }
  };

  render() {
    const { children } = this.props;
    return (
      <Observer onChange={this.loadImage} onlyOnce rootMargin="0% 0% 25%">
        {children}
      </Observer>
    );
  }
}
