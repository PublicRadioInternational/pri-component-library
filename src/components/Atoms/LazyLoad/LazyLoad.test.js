/**
 * @file LazyLoad.test.js
 * Contains tests for LazyLoad.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import LazyLoad from './LazyLoad.component';

describe('<LazyLoad />', () => {
  it('Matches the LazyLoad snapshot', () => {
    const component = renderer
      .create(
        <LazyLoad>
          <span>Lazy Load Me</span>
        </LazyLoad>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
