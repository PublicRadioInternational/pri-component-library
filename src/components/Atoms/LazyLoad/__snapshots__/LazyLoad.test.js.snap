// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyLoad /> Matches the LazyLoad snapshot 1`] = `
<span>
  Lazy Load Me
</span>
`;
